wordpress-mobile/WordPress-iOS
# Offline Mode: Deduplicate new posts

Fixes  pe7hp4-N2-p2#comment-328

## Description 
* Fixes duplicate posts from appearing in the posts list when refresh and sync (upon regaining connectivity) happens in parallel, by sending a UUID as a foreign key in the metadata

## How to test

Precondition: set a response breakpoint on `https://public-api.wordpress.com/rest/v1.2/sites/*/posts/new?context=edit&locale=en`

- Create and save a new draft
- Hit the breakpoint
- ✅ Verify the foreignID is sent
- Pull to refresh
- ✅ Verify that duplicate posts don't appear when refresh and sync (upon regaining connectivity) happens in parallel
- Release the breakpoint

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/99d7c180-10f5-42fb-9d39-0ba66691e81d

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
